internal/sdk/cloudian/sdk.go

```diff
@@ -391,18 +385,24 @@ func (client Client) GetGroup(ctx context.Context, groupId string) (*Group, erro
 	}
 }
 
-func (client Client) newRequest(ctx context.Context, url string, method string, body []byte) (*http.Request, error) {
+func (client Client) newRequest(ctx context.Context, method string, url string, query map[string]string, body []byte) (*http.Request, error) {
 	var buffer io.Reader = nil
 	if body != nil {
 		buffer = bytes.NewBuffer(body)
 	}
-	req, err := http.NewRequestWithContext(ctx, method, url, buffer)
+	req, err := http.NewRequestWithContext(ctx, method, client.baseURL+url, buffer)
 	if err != nil {
 		return req, fmt.Errorf("error creating request: %w", err)
 	}
 	req.Header.Set("Accept", "application/json")
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Set("Authorization", client.authHeader)
 
+	q := req.URL.Query()
+	for k, v := range query {
+		q.Set(k, v)
+	}
+	req.URL.RawQuery = q.Encode()
```

Comment:
Wouldn't it be better to do this before invoking `NewRequestWithContext`? To avoid mutating the request.

Reply:
Agree, but the `url` param is at that point just a string, so we sould have to create the query ourselves. Would think that would be possible!

Reply:
Hmm, we must then concat it onto the url string, such as:

```
q := url.Values{}
for k, v := range query {
	q.Set(k, v)
}

http.NewRequestWithContext(ctx, method, client.baseURL+path+q.Encode()
```

Reply:
Should then use `u, err := urlpkg.Parse(url)` to parse `client.baseURL+path` and then add `q.Encode()` onto that object, and finally stringify the whole thing?

Reply:
Ok, I see now that this will require more code. Ok, Go is Go...

Reply:
https://github.com/statnett/provider-cloudian/commit/26af51a820f4a6f1593475c8a6f84bfb177b96c4